src/Servant/QuickCheck/Internal/HasGenRequest.hs

```diff
@@ -91,9 +91,9 @@ instance (KnownSymbol x, Arbitrary c, ToHttpApiData c, HasGenRequest b)
     genRequest _ = do
       new' <- new
       old' <- old
-      return $ \burl -> let r = old' burl in r {
-          queryString = queryString r
-                     <> param <> "=" <> cs (toQueryParam new') }
+      return $ \burl -> let r = old' burl
+                            qs = queryString r in r {
+          queryString = (if BS.null qs then "" else "&") <> qs <> param <> "=" <> cs (toQueryParam new') }
```

Comment:
Thanks. FWIW I was trying to preserve the `QueryParam` order, will try another way